### PR TITLE
npo:recents

### PR DESCRIPTION
**Pull request by jhoek**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Retrieves only the most recent episodes of the program in question (hence the name). Some programs have so many episodes available that it doesn't make any practical sense to retrieve all, as discussed in issue #7947.

### CODE REVIEW

**Comment by jhoek** [comment]
@dstftw Sorry, this is all a bit new to me - is there anything I can (or need to) do to facilitate the merge process (apart from just being patient)? :) Some of the CI builds fail, but AFAICS not as a result of my code.

File: youtube_dl/extractor/npo.py [hunk 1]
```diff
@@ -478,6 +478,80 @@ class HetKlokhuisIE(NPODataMidEmbedIE):
     }
 
 
+class NPORecentsIE(NPOIE):
+    IE_NAME = 'npo.nl:recents'
+    npo12_regex = r"""<div class='span4'>\s*<div class='image-container'>\s*<a href="(.*?)">\s*(<div class="program-not-available">)?"""
+    npo3_regex = r"""<div class='span4 image'>\s*<a href="(.*?)">\s*<div class="meta-container">\s*<div class="meta first">\s*<div class="md-label"><span class="npo-glyph triangle-right"></span></div>\s*<div class="md-value">.*?</div>\s*</div>\s*</div>\s*(<div class="program-not-available">)?"""
```
Comment (dstftw) [inline youtube_dl/extractor/npo.py]: Both used only once, move to the place where used. Also relax both regexes.

File: youtube_dl/extractor/npo.py [hunk 2]
```diff
@@ -478,6 +478,80 @@ class HetKlokhuisIE(NPODataMidEmbedIE):
...
+    _VALID_URL = r'(?:https?://)?(?:www\.)?npo\.nl/(?P<alt_id>[^/]+)/(?P<program_id>\w+_\d+)'
```
Comment (dstftw) [inline youtube_dl/extractor/npo.py]: Should be `https?://`.

File: youtube_dl/extractor/npo.py [hunk 3]
```diff
@@ -478,6 +478,80 @@ class HetKlokhuisIE(NPODataMidEmbedIE):
...
+            'description': 'md5:a8b6e9d3e3bd367be88766e3ce8e8362'
+        },
+        'playlist_maxcount': 1
+    }]
+
+    def _extract_entries(self, webpage, program_id, program_url):
+        is_npo3 = 'www-assets.npo.nl/uploads/tv_channel/265/logo/smaller_npo3-logo.png' in webpage
+
+        if is_npo3:
+            episodes_url = '%s//search?category=broadcasts&page=1' % program_url
```
Comment (dstftw) [inline youtube_dl/extractor/npo.py]: There are more pages. All should be extracted and extractor should be renamed to `NPOProgramIE`.

File: youtube_dl/extractor/npo.py [hunk 4]
```diff
@@ -478,6 +478,80 @@ class HetKlokhuisIE(NPODataMidEmbedIE):
...
+        'info_dict': {
+            'title': 'Midsomer murders',
+            'id': 'POW_00828660',
+            'description': 'md5:a8b6e9d3e3bd367be88766e3ce8e8362'
+        },
+        'playlist_maxcount': 1
+    }]
+
+    def _extract_entries(self, webpage, program_id, program_url):
+        is_npo3 = 'www-assets.npo.nl/uploads/tv_channel/265/logo/smaller_npo3-logo.png' in webpage
```
Comment (dstftw) [inline youtube_dl/extractor/npo.py]: This is very brittle heuristic.

**Pull request closed without merging**